### PR TITLE
Bump httpclient allowed payload limit

### PR DESCRIPTION
**Pull request by qwertyuu**
## Goal

Bring the upload limit closer to the actual Bugsnag's error reporting limit, see: https://github.com/bugsnag/bugsnag-dotnet/blob/96360fb0f8d31f5fd6a8f3f17184bd626421820e/src/Bugsnag/Payload/Report.cs#L16

from `bugsnag-dotnet`. Also here in the documentation: https://docs.bugsnag.com/platforms/dotnet/asp-net/configuration-options/#maximum-breadcrumbs

### Changed

Changed `MAX_SIZE` const in `HttpClient` to 1024*1024, or 1048576.

## Tests

Used the new constant to ensure that "massive payloads" are properly tested if the value ever changes.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
@kattrali I wonder if the payload limit was 512kb in 2015, which would explain why the code is like this?

File: src/HttpClient.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ class HttpClient
      *
      * @var int
      */
-    const MAX_SIZE = 500000;
+    const MAX_SIZE = 1048576;
```
Comment (kattrali) [inline src/HttpClient.php]: This could instead be the calculation for the value, so its obvious at first glance what is happening here.

```suggestion
    const MAX_SIZE = 1024 * 1024;
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
 Changelog
 =========
 
+## 3.17.1 (2019-08-21)
```
Comment (kattrali) [inline CHANGELOG.md]: No need to add a date, we put it in when we do the release.
```suggestion
## TBD
```

File: tests/HttpClientTest.php [hunk 3]
```diff
@@ -79,7 +79,7 @@ public function testMassiveMetaDataHttpClient()
         $this->guzzle->expects($spy = $this->any())->method('post');
 
         // Add a report to the http and deliver it
-        $this->http->queue(Report::fromNamedError($this->config, 'Name')->setMetaData(['foo' => str_repeat('A', 1000000)]));
+        $this->http->queue(Report::fromNamedError($this->config, 'Name')->setMetaData(['foo' => str_repeat('A', HttpClient::MAX_SIZE * 2)]));
```
Comment (kattrali) [inline tests/HttpClientTest.php]: This test (and the others) should still use a constant value to verify that the proper checks are happening and that `MAX_SIZE` has not been unexpectedly altered.
```suggestion
        $this->http->queue(Report::fromNamedError($this->config, 'Name')->setMetaData(['foo' => str_repeat('A', 1500000)]));
```

**Comment by qwertyuu** [comment]
@kattrali 👍

**Comment by qwertyuu** [comment]
@kattrali 👍

File: src/HttpClient.php (hunk 1 above)
Comment (qwertyuu) [reply src/HttpClient.php]: PHP 5.5 does not seem to like that I use an expression instead of a constant value. https://travis-ci.com/bugsnag/bugsnag-php/jobs/227394423

I'll need to keep it as a whole number for as long as we support that PHP version in the CI.

Comment (GrahamCampbell) [reply src/HttpClient.php]: Yeh, only 5.6+ allow a wider class of expessions as constants, unfortunately.

Comment (kattrali) [reply src/HttpClient.php]: I’d forgotten that it might anger The Ancient Ones. Well maybe add a comment to say it’s a Mb.

**Comment by qwertyuu** [comment]
@kattrali Comment added!

**Comment by kattrali** [comment]
Thanks for the quick edits! ⚡️
